blog/content/post/2021-02-05-testing.md

```diff
@@ -0,0 +1,123 @@
+of automatic availability management make it possible to run both, the single
+tests and complete suites, concerning a set of functionalities that must be
+checked. In result we receive immediate feedback about the changes made and the
+features that were added.
+
+An automated testing infrastructure can be paired with constant integration, and
+delivery pipeline, to enhance the overall performance. We successfully use such
+approach when testing our projects. It all begins when most recent changes for
+the release are ready in our core repository and marked as a
+release. From this point, the automatic process of building usable binaries
+begins, signing it with authenticity proving keys, testing, and delivering the
+final product to the customer, as an online ready-to-download file. Because
+these time-consuming tasks are performing automatically, it leaves developers
+more time for truly creative work on delivering the highest quality of product.
+
+Test verification process needs understanding of the tested functionality and
+further interpretation of the results. In case of the automated testing, the
+output provides result and definition for getting the necessary information for
+users, clients and developers. When the complete regression testing results for
+the given release are ready, the next step is to make them accessible and easy
+to understand. </div>
+
+![testing_chart](/img/test2.png)
+
+#### Why is it so important?
+<div style="text-align: justify">
+Trust and confidence are critical to the success of a firmware product. There
+are two main methods for achieving it: transparency (users can see how the
+testing model is performed) and hard validation (users can track how deeply the
+product is tested). The greatest result brings joining those two methods.
+Publishing the results of hard validation gives the purest quality proof of the
+product deliverable. This is how we are performing in 3mdeb, to could describe our
+products as secure, and optimized.
+
+Usually the testing process is ended when the results are made available
+for public eyes. We could end this post here as well, but there is something
+more. For the analysis it is important, how the tests are
+presented. Usually the results are stored in the charts from publicly available
+spreadsheets, serving the informing role only. This process can be far improved,
+and automated, and this is what we did. While waiting for our automated tests to
+be processed, we have created Dasharo Transparent Validation. System, that
+automatically presents the results in a clean way that is easy to get, and have
+many available filters to customize the view for presenting the exact data
+needed, with support for charts embedabble on a website or repository readme
+notes. You will hear more about Dasharo Transparent Validation soon.
+</div>
+
+![testing_chart](/img/test3.png)
+
+
+## Summary
+
+ We think that our team can help you improving the security of your firmware.
```

Comment:
This is not our standard summary. Secondly we should avoid "we know better what you think" approach. Thirdly we simply can, we have no doubts. Please revert to the old summary.